src/arrayMethodJoin.js

```diff
@@ -4,8 +4,19 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const arrayLength = this.length;
+
+    let convertedString = '';
+
+    for (let i = 0; i < arrayLength; i++) {
+      convertedString += (this[i] === undefined || this[i] === null)
+        ? ''
+        : this[i];
```

Comment:
that's a nice one 👍 
you can improve it by creating separate variable for the condition

Reply:
ok, will do that